force-app/main/default/classes/DonationHistoryController.cls

```diff
@@ -67,24 +63,35 @@ public with sharing class DonationHistoryController {
     * @description method to return Donation History list.
     */
     @AuraEnabled(cacheable=true)
-    public static List<DonationHistoryView> getDonationHistory(Id contactId) {
-        DonationHistoryService service = new DonationHistoryService(contactId);
+    public static List<DonationHistoryView> getDonationHistory(Id contactId, String filter) {
+        DonationHistoryService service = new DonationHistoryService(contactId, filter);
         try {
             List<DonationHistoryView> donationHistories = new List<DonationHistoryView>();
-            service.buildDonationHistory();
-            donationHistories.addAll(buildHistoryFromOCRs(service.getHardCredits(), System.Label.CRLP_HardCredit));
-            donationHistories.addAll(buildHistoryFromOCRs(service.getSoftCredits(), System.Label.CRLP_SoftCredit));
-            donationHistories.addAll(buildHistoryFromOCRs(service.getPartialSoftCredits(), service.getSoftCreditLabel()));
+            donationHistories.addAll(buildHistoryFromOCRs(service.getDonationHistory()));
             return donationHistories;
         } catch(AuraHandledException ex) {
             throw new AuraHandledException(ex.getMessage());
         }
     }
 
-    private static List<DonationHistoryView> buildHistoryFromOCRs(List<OpportunityContactRole> contactRoles, String type) {
+    /**
+     * @description returns the years when the contact made at least 1 donation
+     * @param contactId Id of the contact
+     * @return List<String>
+     */
+    @AuraEnabled(cacheable=true)
+    public static List<String> getYearsWithDonations(Id contactId) {
+        try {
+            return new DonationHistoryService(contactId).getYearsWithDonationsForContact();
+        } catch (Exception e) {
+            throw new AuraHandledException(e.getMessage());
+        }
+    }
+
+    private static List<DonationHistoryView> buildHistoryFromOCRs(List<Opportunity> contactRoles) {
```

Comment:
Likewise, method should be renamed to something like "buildHistoryFromOpportunities"

Reply:
that changed is done in another branch. Its gonna be fixed after merge